clients/src/test/java/org/apache/kafka/common/network/SslTransportLayerTest.java

```diff
@@ -100,6 +140,22 @@ public void testValidEndpointIdentification() throws Exception {
 
         NetworkTestUtils.checkClientConnection(selector, node, 100, 10);
     }
+
+    /**
+     * Tests that hostname verification is performed on the host name or address
+     * specified by the client without using reverse DNS lookup.
+     */
+    @Test
+    public void testEndpointIdentificationNoReverseLookup() throws Exception {
+        String node = "0";
+        server = createEchoServer(SecurityProtocol.SSL);
+        sslClientConfigs.put(SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG, "HTTPS");
+        createSelector(sslClientConfigs);
+        InetSocketAddress addr = new InetSocketAddress("127.0.0.1", server.port());
```

Comment:
I guess here we are trying to test the case that the client connects by ip, but SAN in certificate is hostname? If so, shouldn't we set the server certificate's SAN with hostname before creating createEchoServer?